.github/workflows/ci.yml

```
@@ -322,27 +326,51 @@ jobs:
 
   build_and_test_macos:
     name: Mac
-    runs-on: macos-13
     strategy:
       fail-fast: false
       matrix:
         config: [Release, Debug]
+        os: [macos-13, macos-latest]
+        compiler: [default, clang15]
         include:
+          - config: Release
+            os: macos-13
+            compiler: default
+            coverage: false
           - config: Debug
+            os: macos-13
+            compiler: default
             coverage: true
-            coverage_cxx_flags: "-fprofile-arcs -ftest-coverage" 
+            coverage_cxx_flags: "-fprofile-arcs -ftest-coverage"
             coverage_ld_flags: "-ftest-coverage"
           - config: Release
+            os: macos-latest
+            compiler: clang15
             coverage: false
+          - config: Debug
+            os: macos-latest
+            compiler: clang15
+            coverage: true
+            coverage_cxx_flags: "-fprofile-arcs -ftest-coverage"
+            coverage_ld_flags: "-ftest-coverage"
+        exclude:
+          - os: macos-13
+            compiler: clang15
+          - os: macos-latest
+            compiler: default
+    runs-on: ${{ matrix.os }}
     steps:
       - name: setup
         run: |
-          brew install ninja
-          brew install ccache
+          brew update
+          brew install ninja ccache
       - name: setup coverage
         if: ${{ matrix.coverage }}
+        # needs to be lcov1.14 specifically and brew no longer supports version selection
         run: |
-          brew install lcov
+          git clone --branch v1.14 https://github.com/linux-test-project/lcov.git
+          cd lcov
+          sudo make install
```

Wouldn't this do the trick?

```suggestion
          brew install lcov@1.14
```

did you try that?

---

I did try that - it does not work despite it working for some other formulas. Homebrew versioning has changed a lot over the past few years and while I could use a tap, it's more efficient to get it straight from GitHub.